### ISSUE TITLE
ipv4 dhclient fails to read address

### ISSUE THREAD

**Issue by rueckix** [opnsense/core#5610]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [x] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [x] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

**Describe the bug**

dhclient appears to fail reading the ISP's dhcp offer on boot. There is no known good configuration or opnsense version. It is a first-time setup.

**To Reproduce**

1. Create VLAN nnn for the WAN interface
2. Enable ipv4 dhcp for WAN_VLAN_nnn
3. Reboot opnsense
4. Navigate to Interfaces > Overview > WAN_VLAN_nnn (https://<host>/status_interfaces.php)
5. Observe odd entries as follows: **IPv4 address shows 'dhcp/' (literally)** while the gateway and DNS addresses are correct.

```
IPv4 address: dhcp/
IPv4 gateway: correctly parsed and displayed
DNS servers: correctly parsed and displayed
```

**Expected behavior**
IPv4 address configured as received from the DHCP server.

**Describe alternatives you considered**

* This issue does not self-heal. The above syslog entries repeat every couple of minutes, but the interface remains unconfigured.
* The issue can be rectified by hitting "reload" on the GUI (https://<host>/status_interfaces.php), which appears to restart the dhclient

**Screenshots**

* See screenshot for what is shown on the GUI.
![grafik](https://user-images.githubusercontent.com/39458989/156436777-5178fcf6-6e27-443e-be41-436586b78a87.png)

* See screenshot for what is actually configured:
![grafik](https://user-images.githubusercontent.com/39458989/156434865-ea410195-a371-48d8-bcc6-71ac92e8fab0.png)

**Relevant log files**

From syslog, we see that dhclient actually received an IP address. There are no surrounding error logs.

```
2022-03-02T20:20:03	Notice	dhclient	New Routers (igb2_vlan330): <redacted>	
2022-03-02T20:20:02	Notice	dhclient	New Routers (igb2_vlan330): <redacted>	
2022-03-02T20:20:02	Notice	dhclient	New Broadcast Address (igb2_vlan330): <redacted>	
2022-03-02T20:20:02	Notice	dhclient	New Subnet Mask (igb2_vlan330): <redacted>	
2022-03-02T20:20:02	Notice	dhclient	New IP Address (igb2_vlan330): <redacted>
```

**Additional context**

None

**Environment**

OPNsense 22.1.2-amd64
FreeBSD 13.0-STABLE
OpenSSL 1.1.1m 14 Dec 2021

**Comment by fichtner** [comment]
Have you assigned igb2 parent device and enabled it? It's in the 22.1 release notes.

**Comment by rueckix** [comment]
I confirm. I have assigned the parent device.

**Comment by oli-h** [comment]
Looks like I have the same issue since I updated to OpnSense 22.1:
Specifically:
- I use only one physical interface (igb0) for both LAN (untagged, if-name is `igb0`) and WAN (tagged VLAN 555, if-name is `igb0_vlan555`)
- _note that this setup works in my environment for ~2 years now with OpnSense)_
- `igb0_vlan555` gets IP via DHCP from my Internet-Provider
- Though since OpnSense 22.1 this does not work any more: `igb0_vlan555` simply does not get an IP

To analyze it i did a `tcpdump -i igb0_vlan555` just to see what's going on - but surprisingly I've seen DHCP-requests and -offers (from my provider) and then everything was fine. I continued to test new OpnSense Version 22.1 for a while... didn't realize that my tcpdump was still running in a console

After a few hours I stopped `tcpdump` - then it took only a few seconds and my "internet link" was down again. Hm... Started `tcpdump` again and it worked again.

Long story short: It's the promiscuous mode which did the trick. tcpdump implicit sets interface `igb0_vlan555` to PROMISC-mode and then incoming traffic via VLAN 555 works.
Navigated in OpnSense Web-UI to Interfaces > WAN_555 and switched on _Promiscuous mode_  (which finally is _*permanently* promiscuous mode_) and my WAN-via-VLAN-interface works also without running tcpdump

After some googling I also found a [bug in FreeBSD itself](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=236983). As OpnSense 22.1 switched from HardenedBSD to FreeBSD we now probably also have this problem.

Can you @[rueckix](https://github.com/rueckix) check if  _enabling Promiscuous mode_ solves (or better: works around) your problem, too?

**Comment by rueckix** [comment]
Thanks for the hint. I checked. Promiscuous mode is already enabled on these interfaces.

**Comment by rueckix** [comment]
Update. The same issue is observable in OPNsense 22.1.5-amd64

**Comment by jortkoopmans** [comment]
> Addendum#2: Found hint in OpnSense's 22.1 Changelog (In fact I use MAC-spoofing for my WAN-VLAN-Interface):
> 
> > MAC spoofing now only pertains to the configured interface and not the VLAN siblings or parent interface. This can introduce unwanted configuration due to previous side effects in the code. Make sure to assign and set the spoofed MAC for all interfaces that require a spoofed MAC or simply spoof the MAC on the parent and leave the VLAN sibling settings empty to let them follow the parent MAC automatically. **If in doubt the parent interface can be set into promiscuous mode** now to allow for mixed MAC address use across VLANs too.

Thanks @oli-h  for this, after upgrading (v22.1.6) I also had this issue exactly as described. The `promiscuous mode` override works for me, but has other downsides. My setup involves a VLAN based interface with DHCP client on it. I don't use the interface for anything else than traffic over that VLAN (I use it as WAN), with MAC spoofing.
The best workaround for my situation was the Addendum part: to enable the parent interface and spoof the MAC there, not setting any IP nor use it. Then the VLAN iface can do without `promiscuous mode` and spoofing (it will take the parent MAC).
